### DISCUSSION TITLE
Is there a way to get a return value from user_proxy.initiate_chat() in the Autogen library?

### DISCUSSION THREAD

**Discussion by mapuchill** [microsoft/autogen#104] [Q&A] [answered]
I'm working with the Autogen library to initiate chat sessions and perform various tasks like stock price retrieval. I followed an example found in the project's README to set up the AssistantAgent and UserProxyAgent. However, after executing user_proxy.initiate_chat(), I noticed that the response variable remains empty. The logging only shows what's happening in real-time execution but does not yield a return value that can be stored in a variable for later use.

Here's the example code snippet from the README:
```
import autogen

config_list = autogen.config_list_from_models(model_list=["gpt-4", "gpt-3.5-turbo", "gpt-3.5-turbo-16k"], exclude="aoai")

from autogen import AssistantAgent, UserProxyAgent, config_list_from_json


config_list = config_list_from_json(env_or_file="OAI_CONFIG_LIST")
assistant = AssistantAgent("assistant", llm_config={"config_list": config_list})
user_proxy = UserProxyAgent("user_proxy", code_execution_config={"work_dir": "coding"})

response = user_proxy.initiate_chat(assistant, message="Find NVDA and TESLA stock price change YTD and return it as a json string")
print(response)
```

How can I obtain a return value from user_proxy.initiate_chat() that can be stored in a variable? Is this feature currently not supported, or is there a plan to include this in future releases?

Thanks!

**Comment by 0xRoyalFlush** [comment, marked as answer]
You can retrieve the chat like this:

```
conversation = {}

autogen.ChatCompletion.start_logging(history_dict=conversation)

user_agent.initiate_chat(
ManagerAgent(),
message=query
)

autogen.ChatCompletion.stop_logging()

print(conversation)
```

**Reply by sonichi** [reply]
Yes. You can also access an agent's `chat_messages` or `last_message` property.
https://microsoft.github.io/autogen/docs/reference/agentchat/conversable_agent#chat_messages

**Comment by TEJA2312** [comment]
logging via Completion.start_logging is deprecated in pyautogen v0.2. logging via OpenAIWrapper will be added back in a future release.

**Reply by colaso96** [reply]
Hi @TEJA2312 has this functionality been added back into pyautogen?  this seems like an important feature I would like to add
